utils/dataset_manifest/core.py

```diff
@@ -334,11 +348,13 @@ def __getitem__(self, number):
     def __len__(self):
         return len(self._index)
 
+    def __iter__(self):
+        yield from self._index.values()
```

Comment:
> because self._index can be loaded from a JSON file, the order of entries in which can be arbitrary

It can, but it should not be a problem in realistic scenarios.

Reply:
>  It doesn't use __len__ and requires __getitem__ to throw StopIteration.

`IndexError` also works (and makes more sense for `__getitem__`).

> For me, it looks extra.

What does that mean?

Reply:
> Maybe it shouldn't, but it still could. I think it's worthwhile to defend against rare occurrences.

The index file should only be created by us from the given manifest file. It is always expected to be loaded from a file. If it is invalid for some reason, it means there is either a bug or the user (attacker) tries to hack something.

> `IndexError` also works

Ok, I need to try this option.

Reply:
Done